File: javascript/clock/clock.js
function at(hours, mins) {
  'use strict';
  var time = (60 * hours) + (mins || 0);

  var plus = function(mins) {
    time = (time + mins) % 1440;
    return clock;
  }

  var minus = function(mins) {
    time -= mins;
    // Find better way to wrap around midnight
    while (time < 0) { time += 1440 }
    return clock;
  }

  var equals = function(other) {
    return getTime() === other.getTime();
  }

  var toString = function() {
    return pad(getHours()) + ':' + pad(getMins());
  }

  var pad = function(n) {
    return (n < 10) ? ('0' + n) : n;
  }

  var getHours = function() {
    return parseInt(time / 60);
  }

  var getMins = function() {
    return parseInt(time % 60);
  }

  var getTime = function() {
    return time;
  }

  var clock = {
    plus: plus,
    minus: minus,
    equals: equals,
    getTime: getTime,
    toString: toString
  };

  return Object.create(clock);
}

exports.at = at;

File: python/clock/clock.py
class Clock:
    def __init__(self, hours, mins):
        self.hours = hours
        self.mins = mins
        self.fixup()

    def __eq__(self, other):
        return self.hours == other.hours and self.mins == other.mins

    def __str__(self):
        return (self.format_hours() + ':' +
                self.format_mins())

    def add(self, additional_mins):
        self.mins += additional_mins
        self.fixup()
        return self

    def fixup(self):
        self.fixup_mins()
        self.fixup_hours()

    def fixup_hours(self):
        self.hours = int(self.hours % 24)

    def fixup_mins(self):
        self.hours += self.mins // 60
        self.mins = int(self.mins % 60)

    def format_hours(self):
        return str(self.hours).zfill(2)

    def format_mins(self):
        return str(self.mins).zfill(2)
